client/mysqldump.c

```diff
@@ -4165,6 +4252,397 @@ static char *getTableName(int reset)
+            opt_replace_into ? "/*M!100103 OR REPLACE */ ": "",
+            (strcmp("AGGREGATE", row[2])==0 ? "AGGREGATE " : ""),
+            opt_ignore ? "IF NOT EXISTS " : "", row[0], udf_types[retresult], row[2]);
+  }
+  mysql_free_result(tableres);
+
+  return 0;
+}
+
+
+/*
+  dump all plugins
+*/
+
+static int dump_all_servers()
+{
+  /* No create server yet - MDEV-15696 */
+  MYSQL_ROW row;
+  MYSQL_RES *tableres;
+  MYSQL_FIELD *f;
+  unsigned int num_fields, i;
+  my_bool comma_prepend= 0;
+  const char *qstring;
+
+  if (mysql_query_with_error_report(mysql, &tableres, "SELECT * FROM mysql.servers"))
+    return 1;
+  num_fields= mysql_num_fields(tableres);
+  while ((row= mysql_fetch_row(tableres)))
+  {
+    fprintf(md_result_file,"CREATE %sSERVER %s%s FOREIGN DATA WRAPPER %s OPTIONS (",
+            opt_replace_into ? "/*M!100103 OR REPLACE */ ": "",
+            opt_ignore ? "/*M!100103 IF NOT EXISTS */ " : "", row[0], row[7]);
+    for (i= 1; i < num_fields; i++)
+    {
+      if (i == 7 || row[i][0] == '\0') /* Wrapper or empty string */
+        continue;
+      f= &tableres->fields[i];
+      qstring= (f->type == MYSQL_TYPE_STRING || f->type == MYSQL_TYPE_VAR_STRING) ? "'" : "";
+      fprintf(md_result_file, "%s%s %s%s%s",
+              (comma_prepend ? ", " : ""), f->name, qstring, row[i], qstring);
+      comma_prepend= 1;
+    }
+    fputs(");\n", md_result_file);
+  }
+  mysql_free_result(tableres);
+
+  return 0;
+}
+
+
+/*
+  dump all system statitical tables
+*/
+
+static int dump_all_stats()
+{
+  my_bool prev_no_create_info;
+
+  if (mysql_select_db(mysql, "mysql"))
+  {
+    DB_error(mysql, "when selecting the database");
+    return 1;                   /* If --force */
+  }
+  fprintf(md_result_file,"\nUSE mysql;\n");
+  prev_no_create_info= opt_no_create_info;
+  opt_no_create_info= 1; /* don't overwrite recreate tables */
+  /* EITS added in 10.0.1 */
+  if (mysql_get_server_version(mysql) >= 100001)
+  {
+    dump_table("column_stats", "mysql", NULL, 0);
+    dump_table("index_stats", "mysql", NULL, 0);
+    dump_table("table_stats", "mysql", NULL, 0);
+  }
+  /* Innodb may be disabled */
+  if (!mysql_query(mysql, "show fields from innodb_index_stats"))
+  {
+    MYSQL_RES *tableres= mysql_store_result(mysql);
+    mysql_free_result(tableres);
+    dump_table("innodb_index_stats", "mysql", NULL, 0);
+    dump_table("innodb_table_stats", "mysql", NULL, 0);
+  }
+  opt_no_create_info= prev_no_create_info;
+  return 0;
+}
+
+
+/*
+  dump all system timezones
+*/
+
+static int dump_all_timezones()
+{
+  my_bool opt_prev_no_create_info;
+  if (mysql_select_db(mysql, "mysql"))
+  {
+    DB_error(mysql, "when selecting the database");
+    return 1;                   /* If --force */
+  }
+  opt_prev_no_create_info= opt_no_create_info;
+  opt_no_create_info= 1;
+  fprintf(md_result_file,"\nUSE mysql;\n");
+  dump_table("time_zone", "mysql", NULL, 0);
+  dump_table("time_zone_name", "mysql", NULL, 0);
```

Comment:
I don't get it why you added these tables to &ignore_table in the beginning?

Reply:
per above.